cpp/thirdparty/CMakeLists.txt

```
@@ -51,10 +54,79 @@ endif()
 
 # ## BOOST
 if(MEMILIO_USE_BUNDLED_BOOST)
-    include(BuildBoost)
+    # Ensure that the boost atomic library is used instead of the standard atomic library, where some functionality is only available as of C++20.
+    add_compile_definitions(BOOST_FILESYSTEM_NO_CXX20_ATOMIC_REF)
+    message(STATUS "Downloading Boost library. This may take some time.")
+
+    include(FetchContent)
+    FetchContent_Declare(boost
+    GIT_REPOSITORY https://github.com/boostorg/boost.git
+    GIT_TAG boost-${MEMILIO_BOOST_VERSION})
+    FetchContent_GetProperties(boost)
+
+    if(NOT boost_POPULATED)
+        FetchContent_Populate(boost)
+    endif()
+
+    if(NOT IS_DIRECTORY "${boost_SOURCE_DIR}/boost")
+        if(MSVC)
+            execute_process(COMMAND "bootstrap.bat"
+                WORKING_DIRECTORY ${boost_SOURCE_DIR}
+                )
+            # b2 produces a lot of uninformative output, suppress it.
+            execute_process(COMMAND "b2.exe" headers
+                WORKING_DIRECTORY ${boost_SOURCE_DIR}
+                OUTPUT_QUIET
+                ERROR_QUIET
+                )
+        else()
+            execute_process(COMMAND ./bootstrap.sh
+                WORKING_DIRECTORY ${boost_SOURCE_DIR}
+                )
+            # b2 produces a lot of uninformative output, suppress it.
+            execute_process(COMMAND ./b2 headers 
+                WORKING_DIRECTORY ${boost_SOURCE_DIR}
+                OUTPUT_QUIET
+                ERROR_QUIET
+                )
+        endif()
+    endif()
+    
+    add_library(boost INTERFACE)
+    add_library(Boost::boost ALIAS boost)
+    target_include_directories(boost SYSTEM INTERFACE $<BUILD_INTERFACE:${boost_SOURCE_DIR}> 
+    )

```

```suggestion
    target_include_directories(boost SYSTEM INTERFACE $<BUILD_INTERFACE:${boost_SOURCE_DIR}>)
    if (NOT MSVC)
        target_compile_options(boost INTERFACE "-Wno-c++20-attribute-extensions")
    endif()
```
Since the attribute warnings only come from boost, we should move the new flag in the cpp/CMakeLists.txt to here, something like this should work.

---

I will try if something like this works